torch/quantization/fx/_equalize.py

```diff
@@ -497,21 +502,20 @@ def convert_eq_obs(
             inp_quant_obs_node = node.args[0]
             prev_node = inp_quant_obs_node.args[0]
 
-            # TODO: Possible special handling for connected linear layers
-            # We might need to replace the InputEqualizationObserver with another mul operator...
-            # if prev_node.op == 'call_module' and modules[prev_node.target].dtype != torch.float32:
-            # If this is a connecting linear layer, we want to remove the InputEqualizationObserver
-            #     # Replace the current node with the previous node
-            #     orig_users = list(node.users.keys())
-            #     for user_node in orig_users:
-            #         user_node.replace_input_with(node, prev_node)
-            #     # Erase the node
-            #     model.graph.erase_node(node)
-            #     continue
-
             # Update the following input quantization observer's min/max values
             scale_input_observer(node, modules)
 
+            if prev_node.op == 'call_module' and isinstance(prev_node.target, str) and \
+               isinstance(modules[prev_node.target], nn.Linear):
+                # If this is a connecting linear layer, we want to remove the
+                # InputEqualizationObserver (current node)
+                orig_users = list(node.users.keys())
+                for user_node in orig_users:
+                    user_node.replace_input_with(node, inp_quant_obs_node)
+                # Erase the node
+                model.graph.erase_node(node)
```

Comment:
Can we share this code with other instances where we remove the InputEqualizationObs?